### PR TITLE
[jira/LLC-710]Remove qt dependency in cleanFs

### PR DESCRIPTION
**Pull request by hzheng-ledger**
https://ledgerhq.atlassian.net/browse/LLC-710

### CODE REVIEW

File: core/src/utils/FilesystemUtils.cpp [hunk 1]
```diff
@@ -29,19 +29,31 @@
...
+#include <experimental/filesystem>
 
-void ledger::qt::FilesystemUtils::clearFs(const std::string& path) {
-    QDir root(QString::fromStdString(path));
-    for (const auto& file : root.entryInfoList(QDir::AllEntries | QDir::NoDotAndDotDot)) {
-        if (file.isDir() && file.absoluteFilePath().compare(root.absolutePath()) != 0) {
-            clearFs(file.absoluteFilePath().toStdString());
-        } else if (!file.isExecutable()) {
-            QFile::remove(file.absoluteFilePath());
+using namespace std;
```
Comment (gagbo) [inline core/src/utils/FilesystemUtils.cpp:36]: doesn't look like it is used here

File: core/src/utils/FilesystemUtils.cpp [hunk 2]
```diff
@@ -29,19 +29,31 @@
  *
  */
 
-#include "FilesystemUtils.h"
-#include <QDir>
-#include <QDebug>
-#include <QCoreApplication>
+#include "FilesystemUtils.hpp"
 #include <iostream>
+#include <experimental/filesystem>
```
Comment (gagbo) [inline core/src/utils/FilesystemUtils.cpp:34]: We'll need to make sure this works on all targets, I remember that some experimental features did not compile well on some CI targets

File: core/src/utils/FilesystemUtils.cpp [hunk 3]
```diff
@@ -29,19 +29,31 @@
...
+using namespace std;
+namespace fs = std::experimental::filesystem::v1;
+
+bool ledger::core::FilesystemUtils::isExecutable(const std::string& path){
+    fs::path filePath{path};
+#ifdef _WIN32
+	auto extension=filePath.extension().string();
+    return (extension==".exe") || (extension==".bat") || (extension==".com")
+#else
+    return (status(filePath).permissions() & fs::perms::owner_exec)!=fs::perms::none;
```
Comment (gagbo) [inline core/src/utils/FilesystemUtils.cpp:45]: Only `owner_exec` ? Why not checking `group_exec` and `other_exec` as well ?

File: core/src/utils/FilesystemUtils.hpp [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+ * FilesystemUtils
  * ledger-core
  *
- * Created by Pierre Pollastri on 20/09/2017.
+ * Created by Huiqi ZHENG on 16/12/2016.
  *
  * The MIT License (MIT)
  *
- * Copyright (c) 2017 Ledger
+ * Copyright (c) 2016 Ledger
```
Comment (gagbo) [inline core/src/utils/FilesystemUtils.hpp:10]: Should be 2020

File: core/src/utils/FilesystemUtils.cpp (hunk 2 above)
Comment (hzheng-ledger) [reply core/src/utils/FilesystemUtils.cpp:34]: Yes, I am testing on linux and Mac and Windows on my side to make sure it works.

File: core/src/utils/FilesystemUtils.cpp [hunk 5]
```diff
@@ -29,19 +29,31 @@
...
+void ledger::core::FilesystemUtils::clearFs(const std::string& path) {
+    fs::path filePath{path};
+    for (const auto & file : fs::recursive_directory_iterator(path))
+    {
+        if (!fs::is_directory(file.path())) {
+            if (!FilesystemUtils::isExecutable(file.path().string())){
+        	    fs::remove(file.path());
+            }
         }
     }
```
Comment (gagbo) [inline core/src/utils/FilesystemUtils.cpp]: This loop could also delete the empty directories it meets along the way.

It seems that this loop will only delete non executable files in the given path. It is not consistent with the name of the function

File: core/src/utils/FilesystemUtils.hpp [hunk 6]
```diff
@@ -28,20 +28,22 @@
  * SOFTWARE.
  *
  */
+#ifndef LEDGER_CORE_FILESYSTEMUTILS_HPP
+#define LEDGER_CORE_FILESYSTEMUTILS_HPP
```
Comment (gagbo) [inline core/src/utils/FilesystemUtils.hpp:32]: We use `#pragma once` as much as possible now, so when we change the include guards we change to pragma

Comment (hzheng-ledger) [reply core/src/utils/FilesystemUtils.hpp]: fixed

File: core/src/utils/FilesystemUtils.hpp (hunk 4 above)
Comment (hzheng-ledger) [reply core/src/utils/FilesystemUtils.hpp]: fixed

File: core/src/utils/FilesystemUtils.cpp (hunk 3 above)
Comment (hzheng-ledger) [reply core/src/utils/FilesystemUtils.cpp]: fixed

File: core/src/utils/FilesystemUtils.cpp (hunk 1 above)
Comment (hzheng-ledger) [reply core/src/utils/FilesystemUtils.cpp]: fixed

File: core/src/utils/FilesystemUtils.cpp [hunk 7]
```diff
@@ -42,7 +41,10 @@ bool ledger::core::FilesystemUtils::isExecutable(const std::string& path){
     auto extension=filePath.extension().string();
     return (extension==".exe") || (extension==".bat") || (extension==".com")
 #else
-    return (status(filePath).permissions() & fs::perms::owner_exec)!=fs::perms::none;
+    auto permissions=status(filePath).permissions();
+    return ((permissions & fs::perms::owner_exec)!=fs::perms::none)
+    ||((permissions & fs::perms::group_exec)!=fs::perms::none)
+    ||((permissions & fs::perms::others_exec)!=fs::perms::none);
```
Comment (gagbo) [inline core/src/utils/FilesystemUtils.cpp:47]:
```suggestion
    return (permissions & (fs::perms::owner_exec | fs::perms::group_exec | fs::perms::others_exec))!=fs::perms::none;
```

Should work as well and makes it clearer that we test for any executable flag

File: core/src/utils/FilesystemUtils.cpp (hunk 5 above)
Comment (hzheng-ledger) [reply core/src/utils/FilesystemUtils.cpp]: after some discussion offline. we keep this for this moment

File: core/src/utils/FilesystemUtils.cpp (hunk 2 above)
Comment (hzheng-ledger) [reply core/src/utils/FilesystemUtils.cpp:34]: tested and some small things are fixed in Cmake file

File: core/src/utils/FilesystemUtils.cpp [hunk 8]
```diff
@@ -3,7 +3,7 @@
  * FilesystemUtils.cpp
```
Comment (hlafet-ledger) [inline core/src/utils/FilesystemUtils.cpp:3]: why moving files? why not keeping the old qt function and creating a new .cpp and .hpp file for the new one

File: core/src/utils/FilesystemUtils.hpp [hunk 9]
```diff
@@ -28,20 +28,18 @@
...
-
-#include <string>
+#include <iostream>
 
 namespace ledger {
-    namespace qt {
+    namespace core {
         class FilesystemUtils {
         public:
+        	static bool isExecutable(const std::string& path);
```
Comment (hlafet-ledger) [inline core/src/utils/FilesystemUtils.hpp:41]: indentation

Comment (hzheng-ledger) [reply core/src/utils/FilesystemUtils.hpp]: fixed

File: CMakeLists.txt [hunk 10]
```diff
@@ -29,7 +29,7 @@ set(BOOST_INCLUDEDIR "${CMAKE_CURRENT_SOURCE_DIR}/core/lib/boost/")
 # Notes:
 # > This is a "blind" fix, no available 10.10.x macOS machine,
 # > Issue is specific to 10.10.x, 10.9.5 and > 10.10.x are fine
-set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE STRING "Minimum OS X version to target for deployment: 10.9" FORCE)
+set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "Minimum OS X version to target for deployment: 10.15" FORCE)
```
Comment (hlafet-ledger) [inline CMakeLists.txt:32]: Are supported OS X limited now to 10.15 or earlier ? Would this exclude some users from 10.9 to 10.14 ?

**Review by hlafet-ledger — APPROVED** [review]

**Pull request closed without merging**